README.md

```diff
@@ -1,54 +1,236 @@
 
-    ```shell
-    # create a new Gatsby site using the minimal TypeScript starter
-    npm init gatsby
-    ```
+### Pull Requests
 
-2.  **Start developing.**
+Code changes can be merged into the `develop` branch.
 
-    Navigate into your new site’s directory and start it up.
+Pull request templates are set up on GitHub. In general, PRs can only be merged when:
 
-    ```shell
-    cd my-gatsby-site/
-    npm run develop
-    ```
+- [ ] Someone other than the author has reviewed and approved the changes
+- [ ] All the build checks from Amplify has completed successfully
+- [ ] The branch is synched with the latest changes from the `develop` branch **(optional)**
 
-3.  **Open the code and start customizing!**
+### :palm_tree: Branching Rules
 
-    Your site is now running at http://localhost:8000!
+Whenever a new feature or bugfix is added, a new branch is added, with `develop` as the base branch. For e.g.,
 
-    Edit `src/pages/index.tsx` to see your site update in real-time!
+```
+main
+ develop
+   |
+   |-- feature/add-initial-home-screen
+   |-- fix/cart-button-alignment
+```
 
-4.  **Learn more**
+- use `feature` for new features
+- use `fix` for bugfixes
+- use `refactor` for structure/logic changes
+- use `docs` for documentation changes
+- use `chore` for non-code changes
 
-    - [Documentation](https://www.gatsbyjs.com/docs/?utm_source=starter&utm_medium=readme&utm_campaign=minimal-starter-ts)
+### :closed_book: Style Guide
 
-    - [Tutorials](https://www.gatsbyjs.com/tutorial/?utm_source=starter&utm_medium=readme&utm_campaign=minimal-starter-ts)
+```tsx
+import React from "react"
+import "./ExampleComponet"
 
-    - [Guides](https://www.gatsbyjs.com/tutorial/?utm_source=starter&utm_medium=readme&utm_campaign=minimal-starter-ts)
+import { GetSubscriptions } from "../../utils.ts"
+import { DATE_FORMAT } from "../types.ts"
+import { useSubscription } from "./hooks.ts"
 
-    - [API Reference](https://www.gatsbyjs.com/docs/api-reference/?utm_source=starter&utm_medium=readme&utm_campaign=minimal-starter-ts)
+const DATE_FORMAT = "YYYY-MM-DD"
 
-    - [Plugin Library](https://www.gatsbyjs.com/plugins?utm_source=starter&utm_medium=readme&utm_campaign=minimal-starter-ts)
+const ExampleComponent: React.FC = () => {
+  return (
+    <>
+      <h1>Example Component</h1>
+    </>
+  )
+}
 
-    - [Cheat Sheet](https://www.gatsbyjs.com/docs/cheat-sheet/?utm_source=starter&utm_medium=readme&utm_campaign=minimal-starter-ts)
+export default ExampleComponent
+```
 
-## 🚀 Quick start (Gatsby Cloud)
+#### :card_index_dividers: Folder Structure
 
-Deploy this starter with one click on [Gatsby Cloud](https://www.gatsbyjs.com/cloud/):
+Grouped per component, with global styles, utils, and types in the `src` folder.
 
-[<img src="https://www.gatsbyjs.com/deploynow.svg" alt="Deploy to Gatsby Cloud">](https://www.gatsbyjs.com/dashboard/deploynow?url=https://github.com/gatsbyjs/gatsby-starter-minimal-ts)
+```bash
+node_modules/
+src/
```

Comment:
Small comment here, the folder structure in the best practises doc is a little different. Specifically types, utility and hook files have there own global folder as they are not necessary component specific. Let me know if you disagree here?

Reply:
Oh yeah, my mistake :) 

We can have them as in the doc, and maybe if the stuff gets too convoluted, we can split it up into their own files